sharpy/knowledges/knowledge_test.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+from unittest import mock
+
+from sharpy.knowledges import Knowledge
+from sharpy.managers import DataManager, ManagerBase
+
+
+class TestDataManager(DataManager):
+    def should_be_true(self) -> bool:
+        return self.debug is False
+
+
+class CustomTestManager(ManagerBase):
+    @property
+    def test_property(self) -> int:
+        return 1
+
+    async def update(self):
+        pass
+
+    async def post_update(self):
+        pass
+
+
+class TestKnowledge:
+    @pytest.mark.asyncio
+    async def test_get_DataManager(self):
+        knowledge = Knowledge()
+        knowledge.set_managers(None)
+
+        data_manager = knowledge.get_manager(DataManager)
+
+        assert data_manager is not None
+        assert isinstance(data_manager, DataManager)
+
+    @pytest.mark.asyncio
+    async def test_get_DataManager_from_override(self):
+        knowledge = Knowledge()
+        knowledge.data_manager = TestDataManager()
+        knowledge.set_managers(None)
+
+        data_manager = knowledge.get_manager(DataManager)
+
+        assert data_manager is not None
+        assert isinstance(data_manager, TestDataManager)
+        assert isinstance(data_manager, DataManager)
+
+    @pytest.mark.asyncio
+    async def test_get_TestDataManager_from_override(self):
+        knowledge = Knowledge()
+        knowledge.data_manager = TestDataManager()
+        knowledge.set_managers(None)
+
+        data_manager = knowledge.get_manager(TestDataManager)
+
+        assert data_manager is not None
+        assert isinstance(data_manager, TestDataManager)
+        assert isinstance(data_manager, DataManager)
+        assert data_manager.should_be_true()
+
+    @pytest.mark.asyncio
+    async def test_get_CustomManager_from_override(self):
+        knowledge = Knowledge()
+        knowledge.set_managers([CustomTestManager()])
```

Comment:
Requiring a list to set a single manager is a bit clumsy.

I think that we should provide a method to set to set a single manager.

Reply:
The actual intended usage is supposed to be done in `KnowledgeBot` `configure_managers`. Testing was easier when using the exact `set_managers` method though. Should probably rename that to `_set_managers` and add a docstring warning.